### PR TITLE
Update configure-unit-tests-by-using-a-dot-runsettings-file.md

### LINKED ISSUE #7382: TargetFrameworkVersion entries for dotnet core/5/6 are missing
[Enter feedback here]

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 21ac034a-a498-f3e3-28d0-175b6fead219
* Version Independent ID: 747194a8-ffcf-60b3-b525-6f8fe00066a2
* Content: [Configure unit tests with a .runsettings file - Visual Studio (Windows)](https://docs.microsoft.com/en-us/visualstudio/test/configure-unit-tests-by-using-a-dot-runsettings-file?view=vs-2022)
* Content Source: [docs/test/configure-unit-tests-by-using-a-dot-runsettings-file.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/main/docs/test/configure-unit-tests-by-using-a-dot-runsettings-file.md)
* Product: **visual-studio-windows**
* Technology: **vs-ide-test**
* GitHub Login: @Mikejo5000
* Microsoft Alias: **mikejo**

### PR DESCRIPTION
**Pull request by nohwnd**
Update documentation around .runsettings architecture, tfm and maxCpuCount

Fix #7382

### CODE REVIEW

**Comment by nohwnd** [comment]
> - Did you update the "ms.date" metadata for new or significantly updated articles?
Not sure if this qualifies for the updated date.

**Comment by PRMerger9** [comment]
@nohwnd : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by Court72** [comment]
#label:"aq-pr-triaged"

File: docs/test/configure-unit-tests-by-using-a-dot-runsettings-file.md [hunk 1]
```diff
@@ -180,10 +180,10 @@ The **RunConfiguration** element can include the following elements:
...
+|**MaxCpuCount**|1|**The option name is case sensitive and is easy to mispell as MaxCPUCount**.<br /><br />This setting controls the level of parallelism on process-level. Use 0 to enable the maximum process-level parallelism.<br /><br />This setting determines the maximum number of test dlls, or other test containers that can run in parallel. Each dll will run in its own testhost process, and will be isolated on the process level from the tests in other test dlls. This setting does not force tests in each test dll to run in parallel. Controlling the parallel execution within dll (on thread-level) is up to the test framework such as MSTest, XUnit or NUnit.<br /><br />The default value is `1`, meaning that only 1 testhost will run at the same time. A special value `0` allows as many testhosts as you have logical processors (e.g. 6, for a computer with 6 physical cores without multi-threading, or 12, for a computer with 6 physical cores with multi-threading).<br /><br />The actual number of testhosts that will be started is determined by the amount of distinct dlls in the run.|
```
Comment (Mikejo5000) [inline docs/test/configure-unit-tests-by-using-a-dot-runsettings-file.md:183]:
```suggestion
|**MaxCpuCount**|1|**The option name is case sensitive and is easy to misspell as MaxCPUCount**.<br /><br />This setting controls the level of parallelism on process-level. Use 0 to enable the maximum process-level parallelism.<br /><br />This setting determines the maximum number of test DLLs, or other test containers that can run in parallel. Each DLL will run in its own testhost process, and will be isolated on the process level from the tests in other test DLLs. This setting does not force tests in each test DLL to run in parallel. Controlling the parallel execution within a DLL (on the thread-level) is up to the test framework such as MSTest, XUnit or NUnit.<br /><br />The default value is `1`, meaning that only one testhost will run at the same time. A special value `0` allows as many testhosts as you have logical processors (e.g. 6, for a computer with 6 physical cores without multi-threading, or 12, for a computer with 6 physical cores with multi-threading).<br /><br />The actual number of testhosts that will be started is determined by the amount of distinct DLLs in the run.|
```

File: docs/test/configure-unit-tests-by-using-a-dot-runsettings-file.md [hunk 2]
```diff
@@ -180,10 +180,10 @@ The **RunConfiguration** element can include the following elements:
...
... the exact framework version:<br /><br />For .NET Framework 4.5.1 - 4.8 a testhost that was built with the specified exact version is used. For values outside of that range .NET Framework 4.5.1 testhost is used.<br /><br />For .NET, the actual version is determined by the test project's `<TargetFramework>` (or more precisely `runtimeconfig.json`).<br /><br />For UWP, the test project application is a testhost by itself, and determines the actual version of UWP that is used.<br /><br />Omit the `TargetFrameworkVersion` element from the *.runsettings* file, to automatically determine the framework version from the built binaries.<br /><br />When auto-detecting all target frameworks will be unified into a single common framework. When different version from the same target framework family are found the newer version is chosen (e.g. net452, net472, net48 = net48).<br /><br />When target frameworks from different framework families are found in the run.<br /><br />For .NET Framework runner (in Visual Studio, or vstest.console.exe in Developer command line) the common target framework is to net40. For .NET runner (dotnet test + dlls) the common target framework is set to netcoreapp1.0.|
```
Comment (Mikejo5000) [inline docs/test/configure-unit-tests-by-using-a-dot-runsettings-file.md:185]:
```suggestion
|**TargetFrameworkVersion**| net40 or netcoreapp1.0 |**Omit this whole tag to auto-detect.**<br /><br />This setting defines the framework version, or framework family to use to run tests.<br /><br />Accepted values are any framework moniker such as `net48`, `net472`,`net6.0`, `net5.0`, `netcoreapp3.1`, `uap10.0` or any valid full framework name such as`.NETFramework,Version=v4.7.2` or `.NETCoreApp,Version=v6.0.0`. For backwards compatibility `Framework35`, `Framework40`, `Framework45`, `FrameworkCore10`, `FrameworkUap10` are accepted, meaning (`net35`, `net40`, `net45`, `netcoreapp1.0` and `uap10.0` respectively). All the values are case-insensitive.<br /><br />The provided value is used to determine the test runtime provider to be used. Every test runtime provider must respect the framework family to be used, but might not respect the exact framework version:<br /><br />For .NET Framework 4.5.1 - 4.8 a testhost that was built with the specified exact version is used. For values outside of that range, .NET Framework 4.5.1 testhost is used.<br /><br />For .NET, the actual version is determined by the test project's `<TargetFramework>` (or more precisely `runtimeconfig.json`).<br /><br />For UWP, the test project application is a testhost by itself, and determines the actual version of UWP that is used.<br /><br />Omit the `TargetFrameworkVersion` element from the *.runsettings* file to automatically determine the framework version from the built binaries.<br /><br />When auto-detecting, all target frameworks will be unified into a single common framework. When a different version from the same target framework family is found, the newer version is chosen (e.g. net452, net472, net48 = net48).<br /><br />When target frameworks from different framework families are found in the run.<br /><br />For .NET Framework runner (in Visual Studio, or vstest.console.exe in Developer command line) the common target framework is to net40. For .NET runner (dotnet test + DLLs), the common target framework is set to netcoreapp1.0.|
```

Comment (Mikejo5000) [reply docs/test/configure-unit-tests-by-using-a-dot-runsettings-file.md:185]: @nohwnd Jakub, there is one line that is incomplete: `<br /><br />When target frameworks from different framework families are found in the run.` Should this line be removed?

File: docs/test/configure-unit-tests-by-using-a-dot-runsettings-file.md [hunk 3]
```diff
@@ -180,10 +180,10 @@ The **RunConfiguration** element can include the following elements:
...
+|**TargetPlatform**|x86|**Omit this whole tag to auto-detect.**<br /><br />This setting defines the architecture to use to run tests. Possible values are `x86`, `x64`, `ARM`, `ARM64`, `S390x`.<br /><br />When auto-detecting, the architecture for AnyCPU dlls may differ based on the runner. For .NET Framework runner (in Visual Studio, or vstest.console.exe in Developer command line) the default is x86. For .NET runner (dotnet test) the default is the current process architecture.<br /><br />|
```
Comment (Mikejo5000) [inline docs/test/configure-unit-tests-by-using-a-dot-runsettings-file.md:186]:
```suggestion
|**TargetPlatform**|x86|**Omit this whole tag to auto-detect.**<br /><br />This setting defines the architecture to use to run tests. Possible values are `x86`, `x64`, `ARM`, `ARM64`, `S390x`.<br /><br />When auto-detecting, the architecture for AnyCPU DLLs may differ based on the runner. For .NET Framework runner (in Visual Studio, or vstest.console.exe in Developer command line), the default is x86. For .NET runner (dotnet test), the default is the current process architecture.<br /><br />|
```

File: docs/test/configure-unit-tests-by-using-a-dot-runsettings-file.md [hunk 4]
```diff
@@ -330,16 +330,19 @@ Each element of the file is optional because it has a default value.
 <RunSettings>
   <!-- Configurations that affect the Test Framework -->
   <RunConfiguration>
+    <!-- Use 0 for maximum process-level parallelization. This does not force parallelization within the test dll (on thread-level). You can also change it from the Test menu; choose "Run tests in parallel". Unchecked = 1 (only 1), checked = 0 (max). -->
```
Comment (Mikejo5000) [inline docs/test/configure-unit-tests-by-using-a-dot-runsettings-file.md:333]:
```suggestion
    <!-- Use 0 for maximum process-level parallelization. This does not force parallelization within the test DLL (on the thread-level). You can also change it from the Test menu; choose "Run tests in parallel". Unchecked = 1 (only 1), checked = 0 (max). -->
```

**Review by Mikejo5000 — CHANGES_REQUESTED** [review]
@nohwnd Looks good. I made some style guide edits, etc. See the one comment. You can accept changes in GitHub on the Files changed tab.

**Comment by PRMerger17** [comment]
@Mikejo5000 : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger16** [comment]
@Mikejo5000 : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/test/configure-unit-tests-by-using-a-dot-runsettings-file.md [hunk 5]
```diff
@@ -180,10 +180,10 @@ The **RunConfiguration** element can include the following elements:
...
... exact framework version:<br /><br />For .NET Framework 4.5.1 - 4.8 a testhost that was built with the specified exact version is used. For values outside of that range, .NET Framework 4.5.1 testhost is used.<br /><br />For .NET, the actual version is determined by the test project's `<TargetFramework>` (or more precisely `runtimeconfig.json`).<br /><br />For UWP, the test project application is a testhost by itself, and determines the actual version of UWP that is used.<br /><br />Omit the `TargetFrameworkVersion` element from the *.runsettings* file to automatically determine the framework version from the built binaries.<br /><br />When auto-detecting, all target frameworks will be unified into a single common framework. When a different version from the same target framework family is found, the newer version is chosen (e.g. net452, net472, net48 = net48).<br /><br />When target frameworks from different framework families are found in the run.<br /><br />For .NET Framework runner (in Visual Studio, or vstest.console.exe in Developer command line) the common target framework is to net40. For .NET runner (dotnet test + DLLs), the common target framework is set to netcoreapp1.0.|
```
Comment (Mikejo5000) [inline docs/test/configure-unit-tests-by-using-a-dot-runsettings-file.md:185]:
```suggestion
|**TargetFrameworkVersion**| net40 or netcoreapp1.0 |**Omit this whole tag to auto-detect.**<br /><br />This setting defines the framework version, or framework family to use to run tests.<br /><br />Accepted values are any framework moniker such as `net48`, `net472`,`net6.0`, `net5.0`, `netcoreapp3.1`, `uap10.0` or any valid full framework name such as`.NETFramework,Version=v4.7.2` or `.NETCoreApp,Version=v6.0.0`. For backwards compatibility `Framework35`, `Framework40`, `Framework45`, `FrameworkCore10`, `FrameworkUap10` are accepted, meaning (`net35`, `net40`, `net45`, `netcoreapp1.0` and `uap10.0` respectively). All the values are case-insensitive.<br /><br />The provided value is used to determine the test runtime provider to be used. Every test runtime provider must respect the framework family to be used, but might not respect the exact framework version:<br /><br />For .NET Framework 4.5.1 - 4.8 a testhost that was built with the specified exact version is used. For values outside of that range, .NET Framework 4.5.1 testhost is used.<br /><br />For .NET, the actual version is determined by the test project's `<TargetFramework>` (or more precisely `runtimeconfig.json`).<br /><br />For UWP, the test project application is a testhost by itself, and determines the actual version of UWP that is used.<br /><br />Omit the `TargetFrameworkVersion` element from the *.runsettings* file to automatically determine the framework version from the built binaries.<br /><br />When auto-detecting, all target frameworks will be unified into a single common framework. When a different version from the same target framework family is found, the newer version is chosen (e.g. net452, net472, net48 = net48).<br /><br />For .NET Framework runner (in Visual Studio, or vstest.console.exe in Developer command line) the common target framework is to net40. For .NET runner (dotnet test + DLLs), the common target framework is set to netcoreapp1.0.|
```

**Comment by PRMerger-2** [comment]
@Mikejo5000 : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Pull request merged**